### PR TITLE
[AMS-1890] fix improper BadRequest on Renderer

### PR DESCRIPTION
**Pull request by xakep139**
Additionally fix some StyleCop warnings and update some packages

### CODE REVIEW

File: src/VStore.Renderer/Controllers/PreviewController.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public sealed class PreviewController : VStoreController
             {
                 return TooManyRequests(_retryAfter);
             }
-            catch (ArgumentException ex)
+            catch (ObjectElementInvalidTypeException ex)
```
Comment (xakep139) [inline src/VStore.Renderer/Controllers/PreviewController.cs]: Move to custom exception

File: src/VStore/Objects/ObjectsStorageReader.cs [hunk 2]
```diff
@@ -305,9 +306,8 @@ async Task<(bool IsTruncated, int NextVersionIndex, string NextVersionIdMarker)>
 
             if (!(element.Value is IImageElementValue elementValue))
             {
-                throw new ArgumentException(
-                    $"Element with template code '{templateCode}' of object/versionId '{id}/{versionId}' is not an image.",
-                    nameof(templateCode));
+                throw new ObjectElementInvalidTypeException(
```
Comment (xakep139) [inline src/VStore/Objects/ObjectsStorageReader.cs]: Move to custom exception

File: src/VStore/Objects/IObjectsStorageReader.cs [hunk 3]
```diff
@@ -18,6 +18,16 @@ public interface IObjectsStorageReader
...
+
+        /// <summary>
+        /// Get object's image element value
+        /// </summary>
+        /// <param name="id">Object identifier</param>
+        /// <param name="versionId">Version identifier</param>
+        /// <param name="templateCode">Element template code</param>
+        /// <returns>Value of object's image element</returns>
+        /// <exception cref="S3.ObjectNotFoundException">Object, specified version or element with specified template code is not found</exception>
+        /// <exception cref="S3.ObjectElementInvalidTypeException">Specified object element is not an image</exception>
```
Comment (denisivan0v) [inline src/VStore/Objects/IObjectsStorageReader.cs]: 👍

File: src/VStore/VStore.csproj [hunk 4]
```diff
@@ -7,27 +7,27 @@
...
+    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="2.0.2" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.0.2" />
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="2.0.2" />
     <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
+    <PackageReference Include="Polly" Version="6.0.1" />
     <PackageReference Include="Prometheus.Client" Version="1.4.1" />
     <PackageReference Include="RedLock.net" Version="2.0.0" />
-    <PackageReference Include="SixLabors.ImageSharp" Version="1.0.0-dev001544" />
-    <PackageReference Include="SixLabors.ImageSharp.Drawing" Version="1.0.0-dev001544" />
+    <PackageReference Include="SixLabors.ImageSharp" Version="1.0.0-dev001621" />
```
Comment (xakep139) [inline src/VStore/VStore.csproj]: Fixes https://github.com/SixLabors/ImageSharp/issues/646

**Pull request merged**